### PR TITLE
Add base class for data sources, which use IRepository

### PR DESCRIPTION
**Pull request by arturs105**
Since most of data sources (except `UserDataSource`) return multiple objects and use `IRepository`, this could be used as base class for them.

### CODE REVIEW

File: Toggl.Foundation/DataSources/BaseRepositoryDataSource.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+using Toggl.Multivac.Models;
+using Toggl.PrimeRadiant;
+
+namespace Toggl.Foundation
+{
+    public abstract class BaseRepositoryDataSource<TApi, TDatabaseModel, TModel>
+        where TDatabaseModel : IBaseModel, IDatabaseSyncable
+    {
+        protected readonly TApi api;
+        protected IRepository<TDatabaseModel> storage;
```
Comment (willsb) [inline Toggl.Foundation/DataSources/BaseRepositoryDataSource.cs]: `repository`, not `storage`

File: Toggl.Foundation/DataSources/BaseRepositoryDataSource.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        protected BaseRepositoryDataSource(IRepository<TDatabaseModel> storage, TApi api)
+        {
+            Ensure.Argument.IsNotNull(storage, nameof(storage));
+            Ensure.Argument.IsNotNull(api, nameof(api));
+
+            this.storage = storage;
+            this.api = api;
+        }
+
+        protected abstract void Persist(IEnumerable<TModel> objects);
```
Comment (willsb) [inline Toggl.Foundation/DataSources/BaseRepositoryDataSource.cs]: Not every persist method will be based on `IEnumerable<T>`, so I'd:

```csharp
protected abstract void Persist(TModel item);

protected void Persist(IEnumerable<TModel> items)
{
    foreach(item in items)
    {
        Persist(item);
    }
}

```

instead.

**Comment by amulware** [comment]
Is having this base class still useful, with this little functionality?

**Comment by arturs105** [comment]
Originally i thought that it would have more functionality (using api and repositories), however now it is pretty useless.

**Pull request closed without merging**